core/src/display_object/movie_clip.rs

```diff
@@ -1470,12 +1478,25 @@ impl<'gc> MovieClip<'gc> {
 
     /// Determine what the clip's next frame should be.
     fn determine_next_frame(self) -> NextFrame {
-        if self.current_frame() < self.total_frames() {
+        let mc = self.0.read();
+        let mut reader = mc.static_data.swf.read_from(mc.tag_stream_pos);
+
+        // We ignore the frame count from the header, and instead continue
+        // until we reach the end of the stream or a `TagCode::End`.
+        // Flash Player ignores the frame count, and just executes the full
+        // tag stream before returning to the first frame.
+        if !reader.as_slice().is_empty()
+            && reader.read_tag_code().expect("Failed to read tag") != TagCode::End as u16
```

Comment:
this can panic if the swf happens to have 1 byte hanging at the end, instead of 2. I was thinking it'd be better to fallback to either behavior in this case, but no clue what FP does in this case - does it stop or loop or crash.